### PR TITLE
Add device types introduced in API level 31 or Android Tiramisu

### LINKED ISSUE #43: Panic on Android when enumerating devices using CPAL
I'm trying to enumerate the available devices on Android using this code, through `CPAL`:

```Rust
        let host = cpal::default_host();

        if let Ok(dev) = host.devices() {
            for dev in dev {
                log::warn!("DEVICE: {:?}", dev.name());
            }
        }
```

When running on a Pixel 4a 5G, I get this panic:

```thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', /home/attila/.cargo/registry/src/github.com-1ecc6299db9ec823/oboe-0.4.5/src/java_interface/devices_info.rs:64:18```

cc: https://github.com/torokati44/ruffle-android/issues/4

### PR DESCRIPTION
**Pull request by torokati44**
Source: https://developer.android.com/reference/android/media/AudioDeviceInfo#constants_1

Although I haven't tested it, this might fix #43.

### CODE REVIEW

**Comment by torokati44** [comment]
Although given that this `enum` is even marked `#[non_exhaustive]`, that `unwrap` in `devices_info.rs` should perhaps be replaced with an `unwrap_or_default`, defaulting to `Unknown`?
Or maybe, since `Unknown` seems like a legitimate return value of `getType`, this might be lying, so instead a dummy variant could be added, something like `Unsupported`, or `Unrecognized`...?

**Comment by katyo** [comment]
@torokati44 Added something like `Unsupported` sounds good for first look (maybe with integer field inside).
Could you update that PR?

**Comment by torokati44** [comment]
Yep, done!

**Comment by torokati44** [comment]
Actually.... I wasn't able to test it right away, only now, and:
`error[E0658]: custom discriminant values are not allowed in enums with tuple or struct variants`
https://github.com/rust-lang/rust/issues/60553

**Comment by torokati44** [comment]
For the moment I just removed the `i32` from inside the `Unsupported` variant, and set it to `-1` instead.

File: src/java_interface/devices_info.rs [hunk 1]
```diff
@@ -9,7 +9,9 @@ use super::{
         call_method_no_args_ret_string, get_context, get_devices, get_system_service,
         with_attached, JNIEnv, JObject, JResult,
     },
-    AudioDeviceDirection, AudioDeviceInfo, Context,
+    AudioDeviceDirection, AudioDeviceInfo,
+    AudioDeviceType::Unsupported,
```
Comment (katyo) [inline src/java_interface/devices_info.rs]: Please import type itself here.

Comment (torokati44) [reply src/java_interface/devices_info.rs]: Sure thing!

**Comment by torokati44** [comment]
Thank you for merging!
Do you perhaps have an estimate on when a new release can be pushed to crates.io?
#43 is kind of a blocker for me... :/

**Comment by katyo** [comment]
@torokati44 I published 0.4.6 just now.

**Comment by torokati44** [comment]
Awesome, thanks a lot!

**Pull request merged**